### ISSUE TITLE
Feature Request: Update SONG/SCORE NF processes to read access token from K8s secret. 

### ISSUE THREAD

**Issue by andricDu** [icgc-argo/workflow-roadmap#30]
We will be modifying how the api key / auth creds are passed to the nf tasks, and need to modify song/score to read them.

## Detailed Description
Rather than providing the API Key or JWT in plain text as part of the workflow params, we will be using kubernetes secrets that are mounted inside the task pods. These secrets are generated from workflow-management at schedule time of workflow run. 

## Possible Implementation
- use pod directive in song/score nf to declare where secret is mounted to
- use runId (argo) / runName (nf) as k8s secret name

**Comment by andricDu** [comment]
Adding this to sprint.

**Comment by rosibaj** [comment]
- needs to be tested before merge 
- these are imported into other wfs by a version number
- can make a small wf that uses this version of the tool [submodule]

Once done:
 -run a WF in QA and confirm that it can be completed with the new token changes
- run a WF and interupt it (cancel/kill pod).  Resume the WF and verify that on resume the WF is picking up where it left off, not starting from the beginning.  
- @Buwujiu has a test workflow to use with this (should be on qa and dev)

**Comment by andricDu** [comment]
This was tested inside a local kubernetes environment w/ minikube talking to production SONG in ICGC-DCC. I found a bug and have the fix here: https://github.com/icgc-argo/nextflow-data-processing-utility-tools/pull/20, once @lepsalex approves I will merge.

@rosibaj Once that is merged I will close this ticket and we can create a new ticket under roadmap for rolling out proper apiKey communication to the prod workflows.

**Comment by andricDu** [comment]
Closing. 

Follow up: https://github.com/icgc-argo/workflow-roadmap/issues/32
